string signal
---
bool success
int32 lvalue
